### ISSUE TITLE
Elasticsearch: No way to specify a source filter or other search options

### ISSUE THREAD

**Issue by zlangbert** [akka/alpakka#591]
In elasticsearch as part of a `search` request you can optionally specify a [source filter](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-request-source-filtering.html). This allows you to limit the data returned in the `_source` field of the response items. Because of the way alpakka builds the request, the user can only specify a `query`, and no other search options. It would be great to allow the user to specify a source filter.

In addition to source there are several other options that can be specified as siblings to the query, a solution for this could be flexible enough to allow specifying those as well.

**Comment by ennru** [comment]
Yes, there are lot of other useful parameters. One possibility could be to add some optional string that is added directly to the request entity in `ElasticsearchSourceLogic.sendScrollScanRequest`.

**Comment by mbknor** [comment]
This is probably fixed in master:
https://github.com/akka/alpakka/commit/97f7e46fe41340e147115bb2680eb270150ac609#diff-a05dc5ecaacea54fb635b007c4260294R37

**Comment by yasha-dev1** [comment]
It seems that the new change is only supporting "_source" filter. I am trying to use "size", "form" and "sort" but it is not working while the "_source" works. 
Am I missing something here or the issue hasn't been completely solved? 
Thank you in advance for your response.

**Comment by jmerljak** [comment]
IIUC "from" and "size" are irrelevant as messages are retrieved by scroll scan (see [docs](https://doc.akka.io/docs/alpakka/current/elasticsearch.html#source-configuration)). Results can be limited by Akka Stream [take](https://doc.akka.io/docs/akka/current/stream/operators/Source-or-Flow/take.html) operator.

However, "sort" would be nice.

**Comment by ennru** [comment]
The sort order is now supported with #2654.
